elide-core/src/main/java/com/yahoo/elide/core/datastore/inmemory/InMemoryStoreTransaction.java

```diff
@@ -71,6 +74,80 @@ public InMemoryStoreTransaction(DataStoreTransaction tx) {
         this.tx = tx;
     }
 
+    @Override
+    public Object getRelation(DataStoreTransaction relationTx,
+                              Object entity,
+                              Relationship relationship,
+                              RequestScope scope) {
+        DataFetcher fetcher = new DataFetcher() {
+            @Override
+            public Object fetch(Optional<FilterExpression> filterExpression,
+                                Optional<Sorting> sorting,
+                                Optional<Pagination> pagination,
+                                RequestScope scope) {
+
+                return tx.getRelation(relationTx, entity, relationship.copyOf()
+                        .projection(relationship.getProjection().copyOf()
+                                .filterExpression(filterExpression.orElse(null))
```

Comment:
If we are not gonna to change filter/sorting/pagination here, why do we need to make a copy of the projection? Same for `loadObjects(EntityProjection projection, RequestScope scope)`

Reply:
So the filter expression sorting and pagination can change here.  It is not obvious, but this is a function that is called after the in memory store decides whether to pass sorting, fltering, and pagination to the underlying wrapped store.  In the case of filtering, it can be a subset of the filter expression.